### DISCUSSION TITLE
View model for form submissions

### DISCUSSION THREAD

**Discussion by scottmetoyer** [statamic/cms#5793] [Q&A]
What's the best way to transform or append data to form submissions before returning to the view (something like a view model but for forms).

I'd like to perform calculations on submitted form values and then return those calculations along with the form data to the view.

Alternatively, is there any way to access submitted form values in a view model for an entry? I have a view model set up already and could perform my calculations there if I could get to the form values.

**Comment by edalzell** [comment]
A listener can add some data. If you want to add to next page you can flash it

**Comment by scottmetoyer** [comment]
I ended up reworking this to use a controller. I realized I didn't actually need to store the form submissions at all, I was only interested in computing my calculation and showing it back to the user. I already had display logic in a ViewModel, so I moved that into the new controller, added a method to handle the form posts, and now all my display and calculation logic is bundled up in one place. Works great, thanks for the help!
